pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/ChangeRequestConditional.java

```diff
@@ -0,0 +1,347 @@
+    }
+
+    /**
+     * CHANGE_TARGET ant style matching.
+     * @return target
+     * @see SCMHead#getName()
+     */
+    public String getTarget() {
+        return target;
+    }
+
+    @DataBoundSetter
+    public void setTarget(String target) {
+        this.target = target;
+    }
+
+    /**
+     * CHANGE_BRANCH ant style matching.
+     * @return target
+     * @see ChangeRequestSCMHead2#getOriginName()
+     */
+    public String getBranch() {
+        return branch;
+    }
+
+    @DataBoundSetter
+    public void setBranch(String branch) {
+        this.branch = branch;
+    }
+
+    /**
+     * CHANGE_FORK ant style matching.
+     * @return fork
+     * @see SCMHeadOrigin.Fork#getName()
+     */
+    public String getFork() {
+        return fork;
+    }
+
+    @DataBoundSetter
+    public void setFork(String fork) {
+        this.fork = fork;
+    }
+
+    /**
+     * CHANGE_URL regular expression matching.
+     *
+     * Ant style matching doesn't work on url strings due to it being meant for regular paths
+     * and all those extra characters conflicts.
+     *
+     * @return urlX
+     * @see ObjectMetadataAction#getObjectUrl()
+     */
+    public String getUrlX() {
+        return urlX;
+    }
+
+    @DataBoundSetter
+    public void setUrlX(String urlX) {
+        this.urlX = urlX;
+    }
+
+    /**
+     * CHANGE_TITLE ant style matching.
+     * @return title
+     * @see ObjectMetadataAction#getObjectDisplayName()
+     */
+    public String getTitle() {
+        return title;
+    }
+
+    @DataBoundSetter
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    /**
+     * CHANGE_TITLE regular expression matching.
+     * @return titleX
+     * @see ObjectMetadataAction#getObjectDisplayName()
+     */
+    public String getTitleX() {
+        return titleX;
+    }
+
+    @DataBoundSetter
+    public void setTitleX(String titleX) {
+        this.titleX = titleX;
+    }
+
+    /**
+     * CHANGE_AUTHOR ant style matching.
+     * @return authorDisplayName
+     * @see ContributorMetadataAction#getContributor()
+     */
+    public String getAuthor() {
+        return author;
+    }
+
+    @DataBoundSetter
+    public void setAuthor(String author) {
+        this.author = author;
+    }
+
+    /**
+     * CHANGE_AUTHOR_DISPLAY_NAME ant style matching.
+     * @return authorDisplayName
+     * @see ContributorMetadataAction#getContributorDisplayName()
+     */
+    public String getAuthorDisplayName() {
+        return authorDisplayName;
+    }
+
+    @DataBoundSetter
+    public void setAuthorDisplayName(String authorDisplayName) {
+        this.authorDisplayName = authorDisplayName;
+    }
+
+    /**
+     * CHANGE_AUTHOR regular expression matching.
+     * @return authorDisplayName
+     * @see ContributorMetadataAction#getContributor()
+     */
+    public String getAuthorX() {
+        return authorX;
+    }
+
+    @DataBoundSetter
+    public void setAuthorX(String authorX) {
```

Comment:
I am not a fan of the `X` suffix here. What does it mean to the user? If we're going with different fields for pattern matches vs string matches (quite reasonably), why not `authorPattern`?